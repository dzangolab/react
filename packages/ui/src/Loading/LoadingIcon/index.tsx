import React from "react";

import "./loadingIcon.css";

const LoadingIcon = (
  style: Pick<React.CSSProperties, "fontSize" | "color">
) => {
  return <div data-testid="loading" className="loading" style={style} />;
};

export default LoadingIcon;
