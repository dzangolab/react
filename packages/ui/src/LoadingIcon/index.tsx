import React from "react";

const LoadingIcon = ({
  fontSize,
  color,
}: Pick<React.CSSProperties, "fontSize" | "color">) => {
  return (
    <div
      data-testid="loading"
      className="loading"
      style={{ fontSize, color }}
    />
  );
};

export default LoadingIcon;
