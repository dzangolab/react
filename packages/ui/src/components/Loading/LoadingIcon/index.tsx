import React from "react";

import "./loadingIcon.css";

const LoadingIcon = () => {
  return <div data-testid="loading" className="loading" />;
};

export default LoadingIcon;
