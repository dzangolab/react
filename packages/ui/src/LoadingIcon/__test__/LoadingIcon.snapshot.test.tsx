import { render } from "@testing-library/react";
import React from "react";
import { expect, test } from "vitest";

import LoadingIcon from "../../LoadingIcon";

test("Component matches snapshot", () => {
  const { container } = render(<LoadingIcon />);
  expect(container).toMatchSnapshot();
});
