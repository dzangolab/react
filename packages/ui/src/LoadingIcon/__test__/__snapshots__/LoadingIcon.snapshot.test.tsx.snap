// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component matches snapshot 1`] = `
<div>
  <div
    class="loading"
    data-testid="loading"
  />
</div>
`;
